src/actions/channels.js

```diff
@@ -1095,7 +1108,7 @@ export function favoriteChannel(channelId) {
 
         Client4.trackEvent('action', 'action_channels_favorite');
 
-        savePreferences(currentUserId, [preference])(dispatch, getState);
+        return savePreferences(currentUserId, [preference])(dispatch, getState);
```

Comment:
We might want to `return await savePreferences(...);` here. Otherwise we're returning a promise that returns a promise instead of just a promise that returns a preference.